### ISSUE TITLE
test_hot_standby is flaky

### ISSUE THREAD

**Issue by arssher** [neondatabase/neon#4475]
```
test_runner/regress/test_hot_standby.py:61: in test_hot_standby
    secondary_cursor.execute("SELECT pg_last_wal_replay_lsn()")
/home/nonroot/.pyenv/versions/3.9.2/lib/python3.9/encodings/utf_8.py:15: in decode
    def decode(input, errors='strict'):
E   Failed: Timeout >1800.0s
```

```
secondary/compute.log
...
2023-06-12 14:51:15.908 GMT [21785] LOG:  invalid record length at 0/14B69F0: wanted 24, got 0
2023-06-12 14:51:15.909 GMT [21785] LOG:  waiting for WAL to become available at 0/1002000
2023-06-12 14:51:20.909 GMT [21785] LOG:  invalid record length at 0/14B69F0: wanted 24, got 0
2023-06-12 14:51:20.909 GMT [21785] LOG:  waiting for WAL to become available at 0/1002000
2023-06-12 14:51:25.913 GMT [21785] LOG:  invalid record length at 0/14B69F0: wanted 24, got 0
...
```

https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4470/5244579101/index.html#suites/950eff205b552e248417890b8b8f189e/f5c43665b5b0e373

**Comment by LizardWizzard** [comment]
Another one: https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4552/5403339013/index.html#suites/950eff205b552e248417890b8b8f189e/d3c7e07a5b8ac6ac/

**Comment by bayandin** [comment]
And another one https://neon-github-public-dev.s3.amazonaws.com/reports/pr-4646/5473410503/index.html#suites/950eff205b552e248417890b8b8f189e/6825a2c7bb320da4

**Comment by alexanderlaw** [comment]
I've managed to reproduce this failure locally on 6fb7edf49 (running tens of iterations with 16 tests in parallel), and what I can see in endpoints/secondary/compute.log is:
```
2024-11-15 08:32:34.427 GMT [972800] FATAL:  terminating walreceiver process due to administrator command
2024-11-15 08:32:34.427 GMT [972798] LOG:  invalid record length at 0/14A80D8: wanted 24, got 0
2024-11-15 08:32:34.427 GMT [972798] LOG:  waiting for WAL to become available at 0/1002000
2024-11-15 08:32:34.427 GMT [972798] LOG:  invalid record length at 0/14A80D8: wanted 24, got 0
2024-11-15 08:32:34.427 GMT [972798] LOG:  waiting for WAL to become available at 0/1002000
...
2024-11-15 08:37:29.657 GMT [972798] LOG:  invalid record length at 0/14A80D8: wanted 24, got 0
2024-11-15 08:37:29.657 GMT [972798] LOG:  waiting for WAL to become available at 0/1002000
2024-11-15 08:37:30.427 GMT [972713] LOG:  received fast shutdown request
```

or:
```
2024-11-15 09:07:44.029 GMT [1351386] LOG:  redo starts at 0/14A80D8
2024-11-15 09:07:44.036 GMT [1351386] LOG:  [NEON_SMGR] libpagestore: connected to 'postgresql://no_user@localhost:20002'
2024-11-15 09:07:44.044 GMT [1351386] LOG:  invalid magic number 0000 in log segment 000000010000000000000001, offset 4972544
2024-11-15 09:07:44.044 GMT [1351387] FATAL:  terminating walreceiver process due to administrator command
2024-11-15 09:07:44.046 GMT [1351386] LOG:  invalid magic number 0000 in log segment 000000010000000000000001, offset 4972544
...
2024-11-15 09:12:39.269 GMT [1351386] LOG:  invalid magic number 0000 in log segment 000000010000000000000001, offset 4972544
2024-11-15 09:12:39.269 GMT [1351386] LOG:  waiting for WAL to become available at 0/14BE047
2024-11-15 09:12:40.052 GMT [1351282] LOG:  received fast shutdown request
```

The last commit I can reproduce this on is 634be4f4e~1. So I think this issue should be closed as fixed by https://github.com/neondatabase/neon/pull/4652.

**Comment by arssher** [comment]
Hey! Thanks, indeed likely it should have fixed it.
